ansible/roles/opensearch/tasks/certs.yml

```diff
@@ -0,0 +1,44 @@
+- name: Ensure host certs directory exists
+  file:
+    path: "{{ opensearch_config_path }}/certs"
+    state: directory
+    owner: "{{ opensearch_podman_user }}"
+    group: "{{ opensearch_podman_user }}"
+    mode: ug=rwx,o=
+
+# Based on https://opensearch.org/docs/latest/security-plugin/configuration/generate-certificates/
+- name: Generate private key
+  command:
+    cmd: openssl genrsa -out root-ca-key.pem 2048
+    chdir: "{{ opensearch_config_path }}/certs"
+    creates: root-ca-key.pem
+  become_user: "{{ opensearch_podman_user }}"
+
+- name: Generate root cert
+  command:
+    cmd: "openssl req -new -x509 -sha256 -key root-ca-key.pem -out root-ca.pem -days {{ opensearch_certs_duration }} -subj '/C=XX'"
+    chdir: "{{ opensearch_config_path }}/certs"
+    creates: root-ca.pem
+  become_user: "{{ opensearch_podman_user }}"
+
+- name: Generate node cert
+  # FIXME: need a way to regenerate a new one on expiry?
+  shell:
+    cmd: |
```

Comment:
Is there any way this can be done with ansible modules rather than a multiline shell command?